### ISSUE TITLE
Replay crashes HSDT on latest version

### ISSUE THREAD

**Issue by 305jeff** [HearthSim/Hearthstone-Deck-Tracker#2670]
If you choose stats/latest replay, there is no data even after completing a game just a moment ago. If you try to adjust the filter, it crashes.

If you go to stats/constructed/matches and click "Replay" for the most recent game, it immediately crashes.

Using the latest version of HSDT.

Thanks for all your hard work! Things seem to be mostly working now.

**Comment by eshlin** [comment]
Seeing this as well. Clicking "Replay" on several of my recent games and also older ones leads to an immediate crash. Also, in case it's helpful, I copied the stack trace below, and also clicked "Send Report" on the Crash Report dialog.

Also, appreciate all your hard work! HDT has been invaluable for helping me both track and improve my Hearthstone play, particular with the replay feature. (I'm proud to be a Patreon supporter.)

```
System.InvalidOperationException: Sequence contains no matching element
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
   at Hearthstone_Deck_Tracker.Replay.ReplayViewer.get_PlayerEntity()
   at Hearthstone_Deck_Tracker.Replay.ReplayViewer.Load(List`1 replay)
   at Hearthstone_Deck_Tracker.Replay.ReplayReader.LaunchReplayViewer(String fileName)
   at Hearthstone_Deck_Tracker.Controls.Stats.Constructed.ConstructedGames.ButtonShowReplay_OnClick(Object sender, RoutedEventArgs e)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs e)
   at System.Windows.Controls.Primitives.ButtonBase.OnClick()
   at System.Windows.Controls.Button.OnClick()
   at System.Windows.Controls.Primitives.ButtonBase.OnMouseLeftButtonUp(MouseButtonEventArgs e)
   at System.Windows.UIElement.OnMouseLeftButtonUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.ReRaiseEventAs(DependencyObject sender, RoutedEventArgs args, RoutedEvent newEvent)
   at System.Windows.UIElement.OnMouseUpThunk(Object sender, MouseButtonEventArgs e)
   at System.Windows.Input.MouseButtonEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   at System.Windows.RoutedEventArgs.InvokeHandler(Delegate handler, Object target)
   at System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
   at System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
   at System.Windows.UIElement.RaiseEventImpl(DependencyObject sender, RoutedEventArgs args)
   at System.Windows.UIElement.RaiseTrustedEvent(RoutedEventArgs args)
   at System.Windows.UIElement.RaiseEvent(RoutedEventArgs args, Boolean trusted)
   at System.Windows.Input.InputManager.ProcessStagingArea()
   at System.Windows.Input.InputManager.ProcessInput(InputEventArgs input)
   at System.Windows.Input.InputProviderSite.ReportInput(InputReport inputReport)
   at System.Windows.Interop.HwndMouseInputProvider.ReportInput(IntPtr hwnd, InputMode mode, Int32 timestamp, RawMouseActions actions, Int32 x, Int32 y, Int32 wheel)
   at System.Windows.Interop.HwndMouseInputProvider.FilterMessage(IntPtr hwnd, WindowMessage msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at System.Windows.Interop.HwndSource.InputFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
```

**Comment by Epix37** [comment]
Likely related to the new shaman hero.

**Comment by Epix37** [comment]
also @eshlin, a little teaser : replays will become a lot more awesome soon! :)

**Comment by Epix37** [comment]
Could you guys try if you still have any issues with this build: https://github.com/Epix37/HDT-Test/releases/tag/v0.15.9pre1 (just extract it anywhere and run it)

**Comment by Epix37** [comment]
This should be fixed with https://github.com/HearthSim/Hearthstone-Deck-Tracker/commit/965d3880db3550b2f6f433289c73587708108c3c

**Comment by eshlin** [comment]
Great, thanks for the quick fix! Just upgraded to 0.15.9 and confirmed that all the replays work for me now. Also, thanks for the teaser about replays becoming more awesome soon - excited to see what's coming!
